test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+                                        $p++;
+                                        It '$_.Name' { $_.Name | Should Be 'one' }
+                                     }
+                                }
+    }
+       
+
+    Context 'Hashtable(Stored in a variable) conversion to  PSCustomObject succeeds (Insertion Order is not retained)' {
+        
+        {
+	           $ht = @{one=1;two=2}
+               $script:x = [pscustomobject]$ht
+        } | Should not throw
+       
+        It '$x is not $null' { $script:x | Should Not Be $null }
+        It '$x type' { $script:x | should BeOfType "System.Management.automation.psobject" }
+   }
+
+
+    It 'Conversion from PSCustomObject to hashtable should fail' {
+           
+           try
+           {
+	           $x = [hashtable][pscustomobject]@{one=1;two=2}
+               Throw "Exception expected, execution should not have reached here"
+           }
+           catch
+           {
+                $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'InvalidCastConstructorException'
+           }
+       }
+       
+       
+    Context 'Conversion of Ordered hashtable to PSCustomObject should succeed' {
+      
+       { $script:x = [pscustomobject][ordered]@{one=1;two=2} } | Should Not Throw
+       It '$x is not $null' { $script:x | Should Not Be $null }
+       It '$x type' { $script:x | should BeOfType "System.Management.automation.psobject" }
+       
+       $p = 0
+       # Checks if the first property is One
+       $script:x.psobject.Properties | foreach-object  `
+                                {               
+                                    if ($p -eq 0)  
+                                    {               
+                                        $p++; 
+                                        It 'Name' { $_.Name | Should Be 'one' }
+                                     }
+                                }
+    }
+       
+
+    Context 'Creating an object of an existing type from hashtable should succeed' {       
+       {
+                $source = @"
+                        public class SampleClass1
+                        {                                                       
+                            public SampleClass1(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass1()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $script:x = [SampleClass1]@{a=10;b=5}            
+       } | Should Not Throw
+       It '$x is not $null' { $script:x | Should Not Be $null }
+       It '$x.a' { $script:x.a | Should Be '10' }
+   }

```

Comment:
this can be as simple as:
```
 IT 'Creating an object of an existing type from hashtable should succeed' {
    $result = [System.Management.Automation.Host.Coordinates]@{X=10;Y=33}
    $result.X | should be 10
 } 
```
the check for a null $result is unneeded as if it _is_ null, then $result.X will not be 10!